validphys2/src/validphys/pineparser.py

```diff
@@ -0,0 +1,316 @@
+        raise YamlFileNotFound(yaml_file)
+    ret = yaml.safe_load(yaml_file.read_text())
+    # Make sure the operations are upper-cased for compound-compatibility
+    ret["operation"] = "NULL" if ret["operation"] is None else ret["operation"].upper()
+    return ret
+
+
+def pineko_yaml(yaml_file, grids_folder, check_grid_existence=True):
+    """Given a yaml_file, returns the corresponding dictionary and grids.
+
+    The dictionary contains all information and we return an extra field
+    with all the grids to be loaded for the given dataset.
+
+    Parameters
+    ----------
+    yaml_file : pathlib.Path
+        path of the yaml file for the given dataset
+    grids_folder : pathlib.Path
+        path of the grids folder
+    check_grid_existence: bool
+        if True (default) checks whether the grid exists
+
+    Returns
+    -------
+    yaml_content: dict
+        Metadata prepared for the FKTables
+    paths: list(list(path))
+        List (of lists) with all the grids that will need to be loaded
+    """
+    yaml_content = _load_yaml(yaml_file)
+
+    # Turn the operands and the members into paths (and check all of them exist)
+    ret = []
+    for operand in yaml_content["operands"]:
+        tmp = []
+        for member in operand:
+            p = grids_folder / f"{member}.{EXT}"
+            if not p.exists() and check_grid_existence:
+                raise GridFileNotFound(f"Failed to find {p}")
+            tmp.append(p)
+        ret.append(tmp)
+
+    return yaml_content, ret
+
+
+def pineko_apfelcomb_compatibility_flags(gridpaths, metadata):
+    """
+    Prepare the apfelcomb-pineappl compatibility fixes by matching the apfelcomb content
+    of the metadata to the grids that are being loaded.
+
+    These fixes can be of only three types:
+
+    - normalization:
+        normalization per subgrid
+
+        normalization:
+            grid_name: factor
+
+    - repetition_flag:
+        when a grid was actually the same point repeated X times
+        NNPDF cfactors and cuts are waiting for this repetition and so we need to keep track of it
+
+        repetition_flag:
+            grid_name
+
+    - shifts:
+        only for ATLASZPT8TEVMDIST
+        the points in this dataset are not contiguous so the index is shifted
+
+        shifts:
+            grid_name: shift_int
+
+    Returns
+    -------
+        apfelcomb_norm: np.array
+            Per-point normalization factor to be applied to the grid
+            to be compatible with the data
+        apfelcomb_repetition_flag: bool
+            Whether the fktable is a single point which gets repeated up to a certain size
+            (for instance to normalize a distribution)
+        shift: list(int)
+            Shift in the data index for each grid that forms the fktable
+    """
+    if metadata.get("apfelcomb") is None:
+        return None
+
+    # Can't pathlib understand double suffixes?
+    operands = [i.name.replace(f".{EXT}", "") for i in gridpaths]
```

Comment:
```suggestion
    operands = [i.with_suffix("").stem for i in gridpaths]
```

Alternatively you can do this. `with_suffix` removes one suffix, `stem` the other (if it exists)

Reply:
Honestly I find the string replacement clearer to say "I'm removing the extension `.pineappl.lz4` completely".